sycl/source/detail/xpti_registry.cpp

```diff
@@ -17,22 +17,15 @@ __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 namespace detail {
 #ifdef XPTI_ENABLE_INSTRUMENTATION
-xpti::trace_event_data_t *
-XPTIRegistry::createTraceEvent(void *Obj, const char *ObjName, uint64_t &IId,
-                               const detail::code_location &CodeLoc,
-                               uint16_t TraceEventType) {
-  std::string Name;
-  if (CodeLoc.fileName()) {
-    Name = std::string(CodeLoc.fileName()) + ":" +
-           std::to_string(CodeLoc.lineNumber()) + ":" +
-           std::to_string(CodeLoc.columnNumber());
-  } else {
-    xpti::utils::StringHelper NG;
-    Name = NG.nameWithAddress<void *>(ObjName, Obj);
-  }
+xpti::trace_event_data_t *XPTIRegistry::createTraceEvent(
+    const void *Obj, const void *FuncPtr, uint64_t &IId,
+    const detail::code_location &CodeLoc, uint16_t TraceEventType) {
+  xpti::utils::StringHelper NG;
+  auto Name = NG.nameWithAddress<void *>(CodeLoc.functionName(),
+                                         const_cast<void *>(FuncPtr));
   xpti::payload_t Payload(
       Name.c_str(), (CodeLoc.fileName() ? CodeLoc.fileName() : ""),
-      CodeLoc.lineNumber(), CodeLoc.columnNumber(), (void *)Obj);
+      CodeLoc.lineNumber(), CodeLoc.columnNumber(), const_cast<void *>(Obj));
```

Comment:
Is it better to update this in XPTI directly, so that it stores a const pointer?